### PR TITLE
fix: Update ApplicationPageId with defaultPageId while fetching applications for homepage 

### LINKED ISSUE #10793: [Bug]: The default page is not selected when the user connects the application to git
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The default page is not selected when the user connects the application to git
image

### Steps To Reproduce

1. Connect your application to git
2. Now from master create your own branch 
3. Now change the page name and change the home page  
4. Merge it back to the master 
5. now when we close and open the page the home page is not selected and the page looks like broken 

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

> While fetching the applications for the homepage using all the orgs we were not updating the applicationPageIds which was resulting in the unexpected state as the client always expects the default ids for all the resources. 

Fixes #10793

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> JUnit test
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationFetcherCEImpl.java [hunk 1]
```diff
@@ -140,11 +144,6 @@
                                         final List<Application> applicationList = new ArrayList<>();
                                         if (!CollectionUtils.isEmpty(applicationCollection)) {
                                             applicationList.addAll(applicationCollection);
-                                            long gitConnectedAppsCount = applicationCollection
-                                                    .stream()
-                                                    .filter(application -> application.getGitApplicationMetadata() != null)
-                                                    .count();
-                                            organization.setGitConnectedApplications(gitConnectedAppsCount);
                                         }
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationFetcherCEImpl.java:147]: Why are we removing this? Are we not going to use this ?

**Review by AnaghHegde — APPROVED** [review]
LGTM

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationFetcherCEImpl.java (hunk 1 above)
Comment (abhvsn) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/ApplicationFetcherCEImpl.java:147]: Yeah as this is useless unless we have the limit which brings in lot of overheads for getApplicatins on home page API, so we are removing this bit

**Pull request merged**